content/blog/sivabala-opsql-opsramps-kalpavriksha-wish-fulfilling-tree.md

```diff
@@ -0,0 +1,194 @@
+
+### Search for resources that I tagged in AWS with AWS tag OwnerName :SivaBalaSubramanian
+```PlainText
+installedAppName = "aws" AND tags.name = "OwnerName" and tags.value = "SivaBalaSubramanian"​
+```
+### Search for resources with alerts that have been open for the last 2 hours​
+
+```PlainText
+createdTime > "-7200sec" ​
+createdTime > "-120min"​
+```
+
+### Search for Open alerts that are Critical​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" ​
+```
+​
+### Search for Open and Critical alerts that have an Open incident​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" AND incidentId IS NOT NULL ​
+```
+
+### Search for alerts that have been Open for longer than 2 hours​
+```PlainText
+createdTime < "-7200sec" ​
+``` 
+​
+### Search for Open and Critical alerts on resources tagged with AWS tag “BU: bu-123”​
+```PlainText
+currentState = "CRITICAL" AND status = "OPEN" AND resource.tags.name = "Team" AND resource.tags.value = "Opsqa"​
+```
+
+## What is OpsQL API?
+The OpsQL API is a powerful interface that allows users to execute OpsQL queries programmatically. This API provides the flexibility to filter, search data within the OpsRamp platform, making it an indispensable tool for IT administrators and developers.
+
+#### Key Features
+
+ 1.  **Comprehensive Querying**: The OpsQL API supports a wide range of query operations, allowing users to filter data based on various attributes and conditions.
+ 2.  **Flexibility**: Users can create complex queries using logical operators and a variety of comparison operators.
+3.  **Integration**: The API can be integrated into custom applications, scripts, and workflows, enhancing automation and efficiency.
+
+
+#### Basic Syntax and Structure
+
+The general structure of an OpsQL API request involves specifying the tenant ID and the query payload. Here’s a basic example:
+
+
+## Basic Syntax and Structure
+The general structure of an OpsQL API request involves specifying the tenant ID and the query payload. Here’s a basic example:
+
+Here’s a basic example:
+
+```
+POST /opsql/api/v3/tenants/{tenantId}/queries
+```
+
+The request body typically includes:
+
+* *   `objectType`: The type of object to query (e.g., resource, alert, ticket).
+* *   `fields`: The fields to retrieve.
+* *   `filterCriteria`: The criteria to filter the objects.
+
+## Common Use Cases and code samples
+
+### Filtering critical alerts
+
+```python
+import requests
+import json
+accessToken='valid access token'
+
+
+url = "https://server/opsql/api/v3/tenants/client_id/queries"
+
+payload = json.dumps({
+  "objectType": "alert",
+  "fields": [
+    "id",
+    "clientId",
+    "component",
+    "currentState"
+  ],
+  "filterCriteria": "currentState=critical"
+})
+headers = {
+  'Accept': 'application/json',
+  'Content-Type': 'application/json',
+  'Authorization': accessToken
+}
+
+response = requests.request("POST", url, headers = headers, data = payload)
+```
+### Saving OpsQL response as CSV file

```

Comment:
It would be nice to have some text underneath this header and before the code that says a little something about this. Perhaps something like:

There are times that you would want to save an OpsQL response as a CSV file, perhaps for (and give an example). This is how that would be done: